dags/pull_here.py

```diff
@@ -9,46 +9,119 @@
 from datetime import datetime, timedelta
 from airflow.operators.bash_operator import BashOperator
+from airflow.models.connection import Connection
+from airflow.hooks.base_hook import BaseHook
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
 from airflow.providers.postgres.hooks.postgres import PostgresHook
-from airflow.models import Variable 
+from airflow.models import Variable
+from airflow.decorators import dag, task, task_group
+from airflow.macros import ds_add, ds_format
 
 try:
     repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
     sys.path.insert(0, repo_path)
     from dags.dag_functions import task_fail_slack_alert
+    from here.traffic.here_api import query_dates, get_access_token, get_download_url, download_data, send_data_to_database
+    cfg_path = repo_path
 except:
     raise ImportError("Cannot import slack alert functions")
 
-dag_name = 'pull_here'
+doc_md = """
+
+### The Daily HERE pulling DAG [probe]
+
+This DAG runs daily to pull here data from traffic analytics' API to here.ta in the bigdata database using Taskflow.
+Slack notifications is raised when the airflow process fails. 
+This pulles probe data in oppose to path in the pull_here_path DAG.
 
+"""
+
+dag_name = 'pull_here'
 dag_owners = Variable.get('dag_owners', deserialize_json=True)
 names = dag_owners.get(dag_name, ['Unknown']) #find dag owners w/default = Unknown    
 
-here_postgres = PostgresHook("here_bot")
-rds_con = here_postgres.get_uri()
-
 default_args = {'owner': ','.join(names),
                 'depends_on_past':False,
                 'start_date': pendulum.datetime(2020, 1, 5, tz="America/Toronto"),
                 'email_on_failure': False,
                 'email_on_success': False,
-                'retries': 3, #Retry 3 times
+                'retries': 0, #Retry 3 times
                 'retry_delay': timedelta(minutes=60), #Retry after 60 mins
                 'retry_exponential_backoff': True, #Allow for progressive longer waits between retries
                 'on_failure_callback': task_fail_slack_alert,
-                'env':{'here_bot':rds_con,
-                       'LC_ALL':'C.UTF-8', #Necessary for Click
+                'env':{'LC_ALL':'C.UTF-8', #Necessary for Click
                        'LANG':'C.UTF-8'}
                 }
 
-dag = DAG(dag_id = dag_name, default_args = default_args, schedule = ' 30 16 * * * ')
-#Every day at 1630
+@dag(dag_id = dag_name,
+     default_args=default_args,
+     schedule_interval='30 16 * * * ' ,
+     catchup=False,
+     doc_md = doc_md,
+     tags=["HERE"]
+     )
+
+def pull_here():
+
+    @task
+    def send_request():
+        api_conn = BaseHook.get_connection('here_api_key')
+        access_token = get_access_token(api_conn.password, api_conn.extra_dejson['client_secret'], api_conn.extra_dejson['token_url'])
+        return access_token
+
+    @task
+    def get_request_id(access_token: str, **kwargs):
```

Comment:
[With taskflow](https://airflow.apache.org/docs/apache-airflow/2.8.2/templates-ref.html#accessing-airflow-context-variables-from-taskflow-tasks) you can just use `ds=None` instead of **kwargs

Reply:
https://airflow.apache.org/docs/apache-airflow/2.5.0/concepts/taskflow.html#context
I only read the first line when writing this lolol ds=None is cleaner, I will switch to that